api/app/registration/api.py

```diff
@@ -1,21 +1,191 @@
+                return errors.FORBIDDEN
+
+            offer.updated_at = datetime.now()
+            offer.accepted = accepted
+            offer.rejected = rejected
+            offer.rejected_reason = rejected_reason
+
+            db.session.commit()
+
+        except:
+            LOGGER.error("Failed to update offer with id {}".format(args['offer_id']))
+            return errors.ADD_OFFER_FAILED
+        return offer_update_info(offer), 201
+
+    @admin_required
+    @marshal_with(offer_fields)
+    def post(self):
+        args = self.req_parser.parse_args()
+        user_id = args['user_id']
+        email = g.current_user['email']
+        event_id = args['event_id']
+        offer_date = datetime.strptime((args['offer_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        expiry_date = datetime.strptime((args['expiry_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        payment_required = args['payment_required']
+        travel_award = args['travel_award']
+        accommodation_award = args['accommodation_award']
+        updated_at = datetime.strptime((args['updated_at']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        user = db.session.query(AppUser).filter(AppUser.email == email).first()
+
+        offer_entity = Offer(
+            user_id=user_id,
+            event_id=event_id,
+            offer_date=offer_date,
+            expiry_date=expiry_date,
+            payment_required=payment_required,
+            travel_award=travel_award,
+            accommodation_award=accommodation_award,
+            updated_at=updated_at
+        )
+
+        db.session.add(offer_entity)
+        db.session.commit()
+
+        if user.email:
+            send_mail(recipient=user.email, subject='Offer from Deep Learning Indaba',
+                      body_text=OFFER_EMAIL_BODY.format(
+                            user.user_title, user.firstname, user.lastname,
+                            user_id, event_id, offer_date,
+                            expiry_date, payment_required,
+                            travel_award, accommodation_award))
+
+            LOGGER.debug("Sent an offer email to {}".format(user.email))
+        else:
+            user.verified_email = True
+            try:
+                db.session.commit()
+            except IntegrityError:
+                LOGGER.error("Unable to verify email: {}".format(email))
+                return errors.VERIFY_EMAIL_OFFER
+        
+        return offer_info(offer_entity), 201
+
+    @auth_required
+    @marshal_with(offer_fields)
+    def get(self):
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+        user_id = g.current_user['id']
+
+        try:
+            offer = db.session.query(Offer).filter(Offer.event_id == event_id).filter(Offer.user_id == user_id).first()
+            if not offer:
+                LOGGER.warn(
+                    "Offer not found for event_id: {}".format(args['event_id']))
+                return errors.EVENT_NOT_FOUND
```

Comment:
Can we use an OFFER_NOT_FOUND error message here please?

Reply:
Yes we can i think... except it won't contain that event id?